core/corehttp/gateway_handler.go

```diff
@@ -334,45 +325,55 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	// set these headers _after_ the error, for we may just not have it
-	// and don't want the client to cache a 500 response...
-	// and only if it's /ipfs!
-	// TODO: break this out when we split /ipfs /ipns routes.
-	modtime := time.Now()
-
-	if f, ok := dr.(files.File); ok {
-		if strings.HasPrefix(urlPath, ipfsPathPrefix) {
-			w.Header().Set("Cache-Control", "public, max-age=29030400, immutable")
-
-			// set modtime to a really long time ago, since files are immutable and should stay cached
-			modtime = time.Unix(1, 0)
+	// Support custom response formats passed via ?format or Accept HTTP header
+	if contentType := getExplicitContentType(r); contentType != "" {
+		switch contentType {
```

Comment:
```suggestion
		switch getExplicitContentType(r) {
		case "":
			break
```
Personally, I find that clearer.

Note (can't commit as is, you need to remove the tail `}` for the removed `if`)